src/Analysis/Ast/Impl/Analyzer/Symbols/ClassEvaluator.cs

```diff
@@ -118,6 +111,40 @@ internal sealed class ClassEvaluator : MemberEvaluator {
             UpdateClassMembers();
         }
 
+        private IEnumerable<IPythonType> ProcessBases(Scope outerScope) {
+            var bases = new List<IPythonType>();
+            foreach (var a in _classDef.Bases.Where(a => string.IsNullOrEmpty(a.Name))) {
+                var expr = a.Expression;
+                var m = Eval.GetValueFromExpression(expr);
+
+                switch (m?.MemberType) {
+                    case PythonMemberType.Method:
+                    case PythonMemberType.Function:
+                    case PythonMemberType.Property:
+                    case PythonMemberType.Instance:
+                    case PythonMemberType.Variable when m is IPythonConstant:
+                        // all invalid types to inherit from
```

Comment:
I think you can derive from instance if the instance is a type info. I.e.
```python
class A: ...
a = A
class B(a): ...
```
appears to work
![image](https://user-images.githubusercontent.com/12820357/61312364-1f6e1680-a7ad-11e9-965a-a9fd8caaef47.png)

so we need to test that

Reply:
Will check.

Reply:
Works fine. The reason is that I resolve the name expression 'a' to a member whose type is Class.